rados/ioctx_set_alloc_hint.go

```diff
@@ -0,0 +1,38 @@
+//go:build ceph_preview
+// +build ceph_preview
+
+package rados
+
+// #cgo LDFLAGS: -lrados
+// #include <rados/librados.h>
+// #include <stdlib.h>
+//
+import "C"
+
+import (
+	"unsafe"
+)
+
+// SetAllocationHint sets allocation hint for an object. This is an advisory
+// operation, it will always succeed (as if it was submitted with a
+// LIBRADOS_OP_FLAG_FAILOK flag set) and is not guaranteed to do anything on
+// the backend.
+//
+// Implements:
+//  void rados_set_alloc_hint2(rados_ioctx_t io,
+//                             const char *o,
+//                             uint64_t expected_object_size,
+//                             uint64_t expected_write_size,
+//                             uint32_t flags);
+func (ioctx *IOContext) SetAllocationHint(oid string, expectedObjectSize uint64, expectedWriteSize uint64, flags AllocHintFlags) error {
+	coid := C.CString(oid)
+	defer C.free(unsafe.Pointer(coid))
+
+	return getError(C.rados_set_alloc_hint2(
```

Comment:
Can this return an error? In the comment you write that it always succeeds.

Reply:
Setting the hint in itself is guaranteed to never fail, but I guess we could have a network or authentication error or any other kind of problem in between ?

Reply:
I generally agree, if the ceph call returns a error value, even if it's currently always zero it is reasonable to wrap it the common way and if ceph ever started returning non-zero errors, we'd already be handling them. It'd be different if we had to do a lot of custom error handling for this function but its "near free" since it's common throughout the codebase.

Reply:
So now I've double checked.  The comment's "Implements" block indicates a void return, but looking at the ceph code in src/include/rados/librados.h the return type is an int. rados_write_op_set_alloc_hint2 returns "void".  Please correct the comment, but leave the go code as-is.